timeseriesx/base/timestamp_series.py

```diff
@@ -392,8 +393,12 @@ def __eq__(self, other):
 
     def __getitem__(self, item):
         if isinstance(item, (slice, Iterable)):
-            new_ts = copy.deepcopy(self)
-            new_ts._series = new_ts._series[item]
+            new_ts = TimestampSeries(
+                series=self._series[item].copy(),
+                unit=self.unit,
+                freq=self.freq,
+                time_zone=self.time_zone,
+            )
```

Comment:
I get your point and i agree. My idea was to benefit from the coercion and validation that is automatically triggered in the constructor, because slicing can break the `freq` attribute, especially when slicing by a list of timestamps. I guess I can trigger it manually instead. 

I still am not completely happy with the validation. The goal of the package is to ensure that transformations result in a valid time series. But by now, it is not ensured everywhere. I think this is also closely related to the design with mixins. I am not too happy with that either. The original idea was to make the features (time zone, unit and frequency) independent from one another, and make them kind of pickable as add-ons for a basic time series. But now I don't see the benefit anymore and I think it makes the code harder to read. It also results in strange code in cases where the features interfere. I really would like to know your opinion on that. What do you think about opening an issue to discuss it?